crates/libs/core/src/runtime/package_change/config.rs

```diff
@@ -0,0 +1,143 @@
+{
+    fn OnPackageAdded(
+        &self,
+        _source: Option<&mssf_com::FabricRuntime::IFabricCodePackageActivationContext>,
+        configpackage: Option<&mssf_com::FabricRuntime::IFabricConfigurationPackage>,
+    ) {
+        let new_package = ConfigurationPackage::from_com(configpackage.unwrap().clone());
+        let event = ConfigurationPackageChangeEvent::Addition { new_package };
+        // TODO: unwrap, or should we change the return type of the lambda to be the empty type?
+        self.inner.on_change(&event).unwrap();
+    }
+
+    fn OnPackageRemoved(
+        &self,
+        _source: Option<&mssf_com::FabricRuntime::IFabricCodePackageActivationContext>,
+        configpackage: Option<&mssf_com::FabricRuntime::IFabricConfigurationPackage>,
+    ) {
+        let previous_package = ConfigurationPackage::from_com(configpackage.unwrap().clone());
+        let event = ConfigurationPackageChangeEvent::Removal { previous_package };
+        self.inner.on_change(&event).unwrap();
+    }
+
+    fn OnPackageModified(
+        &self,
+        _source: Option<&mssf_com::FabricRuntime::IFabricCodePackageActivationContext>,
+        previousconfigpackage: Option<&mssf_com::FabricRuntime::IFabricConfigurationPackage>,
+        configpackage: Option<&mssf_com::FabricRuntime::IFabricConfigurationPackage>,
+    ) {
+        let new_package = ConfigurationPackage::from_com(configpackage.unwrap().clone());
+        let previous_package =
+            ConfigurationPackage::from_com(previousconfigpackage.unwrap().clone());
+        let event = ConfigurationPackageChangeEvent::Modification {
+            previous_package,
+            new_package,
+        };
+        self.inner.on_change(&event).unwrap();
+    }
+}
+
+/// Lambda implementation of ConfigurationPackageChangeEventHandler trait.
+/// This is used in FabricClientBuilder to build function into handler.
+/// Not exposed to user.
+/// Strictly speaking we don't need this layer. But it would allow us to open the door to trait implementations someday
+pub(crate) struct LambdaConfigurationPackageEventHandler<T>
+where
+    T: Fn(&ConfigurationPackageChangeEvent) -> crate::Result<()> + 'static,
+{
+    f: T,
+}
+
+impl<T> LambdaConfigurationPackageEventHandler<T>
+where
+    T: Fn(&ConfigurationPackageChangeEvent) -> crate::Result<()> + 'static,
+{
+    pub fn new(f: T) -> Self {
+        Self { f }
+    }
+}
+
+impl<T> ConfigurationPackageChangeEventHandler for LambdaConfigurationPackageEventHandler<T>
+where
+    T: Fn(&ConfigurationPackageChangeEvent) -> crate::Result<()> + 'static,
+{
+    fn on_change(&self, change: &ConfigurationPackageChangeEvent) -> crate::Result<()> {
+        (self.f)(change)
+    }
+}
+
+/// This struct ensures that the handle is retained and deregistered before the implementation is dropped
+pub struct ConfigurationPackageChangeCallbackHandle {
+    activation_context: IFabricCodePackageActivationContext6,
+    handle: i64,
```

Comment:
Just expose actctx function to return and handle and user needs to unregister. Not deviate too much from the COM api. I think the runtime will auto cleanup the handle at drop time.

Reply:
U can add another type for auto drop. On top of the handle.

Reply:
Better?

Reply:
Separated the behavior.

Reply:
let me know which you'd like

Reply:
Removed auto-drop for now, can always add it later if we want it.